### DISCUSSION TITLE
How to use resolvers in Controllers with React Native

### DISCUSSION THREAD

**Discussion by ZeroPie** [react-hook-form/react-hook-form#3399] [Q&A] [answered]
Hi, its been awesome working with hook forms :)
I did not find any specific documentation on integrating the resolvers with React Native and since i opted to use the Controller i am not sure if i am missing something, because my Validation does not trigger errors.

```
const { control, handleSubmit, errors, reset, formState, register } = useForm(
    {
      resolver: joiResolver(schema)
    }
  )

const schema = Joi.object({
  height_cm: Joi.number(),
  weight_kg: Joi.number()
})
...
        <View style={styles.formRow}>
          <View style={styles.spaceBetweenRow}>
            <Controller
              control={control}
              name='height_cm'
              defaultValue=''
              render={({ onChange, onBlur, value, ...props }) => (
                <View style={styles.flexGrow1}>
                  <TextInput
                    {...props}
                    label='Körpergrösse in cm'
                    error={!!errors.height_cm}
                    onBlur={onBlur}
                    keyboardType='phone-pad'
                    onChangeText={value => onChange(value)}
                    value={value}
                    ref={register}
                    style={styles.grow1Margin20}
                    mode={'outlined'}
                  />
                  <HelperText type='error' visible={!!errors.height_cm}>
                    Requiered
                  </HelperText>
                </View>
              )}
            />
...
```

I also tried making my own custom hook resolver but to no avail. The form works and it wont let me submit, but no errors are triggered.

**Comment by bluebill1049** [comment, marked as answer]
- u should use either resolver or build-in validator 
- you don't need to register again in Controller: https://react-hook-form.com/get-started#ReactNative

**Reply by ZeroPie** [reply]
Thank you! i did oversee that, can i missuse this thread to ask how to only validate dirty/touched inputs?

**Reply by bluebill1049** [reply]
i think you are after mode `touched`?

**Reply by bluebill1049** [reply]
<img width="718" alt="Screen Shot 2020-11-09 at 11 02 04 pm" src="https://user-images.githubusercontent.com/10513364/98538927-98b54500-22df-11eb-8ca7-9ee23910fadb.png">

**Reply by ZeroPie** [reply]
Yes, Thanks!
